### ISSUE TITLE
CTC loss does not go down

### ISSUE THREAD

**Issue by chaoli95** [kilsen512/R3DCNN-tensorflow#3]
Hi. Appreciate you work! I adjust your code for my dataset, but ctc loss doesn't go down, and accuracy remains only 20%. Have you met this problem before? 
Also, I noticed you used your own calculate_label function instead of ctc_greedy_decoder or ctc_beam_search_decoder. I choose ctc_greedy_decoder. Is this the problem? 
Thank you.

**Comment by kilianyp** [comment]
Hi,
it has been some time that I have worked with this, so I am not a 100% sure about everything.
But I remember that the learning rate was really important (as always) and was the main parameter to fine tune.

Regarding the decoder, ctc_greedy decoder is used here [here](https://github.com/kilsenp/R3DCNN-tensorflow/blob/f7fe4a0e8e48055d3185932bce011e24c585d0d1/networks/r3dcnn.py#L138).
This calculate_label is basically another decoding on top of ctc greedy decoding. It assigns the whole clip only one single label.
The greedy decoding is anyway independent of the loss calculation, therefore, it has nothing to do with it not decreasing. Check [here](https://github.com/kilsenp/R3DCNN-tensorflow/blob/f7fe4a0e8e48055d3185932bce011e24c585d0d1/networks/r3dcnn.py#L119).

Good luck.

Maybe one other hint, make sure you are reading your data in correctly!
Back then, I kept confusing batch, time and the image dimensions.
Also the time dimension is limited right now (I think).

**Comment by lchaooo** [comment]
Thank you for replying.
I'm pretty sure the data's dimensions is correct.
Do you remember your learning rate? And can you do me a favor and check tower_loss function in r3dcnn.py? I think it's not correct...

**Comment by chaoli95** [comment]
Thank you for replying.
I have a question for tf.nn.ctc_loss function's parameters. 
The [document](https://www.tensorflow.org/api_docs/python/tf/nn/ctc_loss) said, "The inputs Tensor's innermost dimension size, num_classes, represents num_labels + 1 classes, where num_labels is the number of true labels, and the largest value (num_classes - 1) is reserved for the blank label." I noticed your inputs parameter's innermost dimension size is num_labels + 2 because you set "0" as "no gesture" class. According to my understanding, it's actually the same class because "no gesture" means the that there is a "blank" between two actual gestures, am I right? 

UPDATE: I chose the proper learning rate. IT WORKS! Thank you!

**Comment by buaa-luzhi** [comment]
@lchaooo
I think it's actually the same class because "no gesture" means the that there is a "blank" between two actual gestures. 
Can you show me the structure of the dataset in detail. Furthermore, which benchmark dataset are you using.
Thank you very much!

**Comment by buaa-luzhi** [comment]
@lchaooo
Hello, I have changed the size of the learning rate several times.  .However, the loss function value of the training set is still large.  It starts to decay over 20,000. Have you ever been in this situation before.
Thanks so much!
